src/main.c

```diff
@@ -0,0 +1,49 @@
+/******************************************************
+* Author : Amuleen Gulati 
+* Function : main
+* Application : This function takes sensor values as input and 
+* calls different functions to produce an output
+* Input : a .csv file containing temperature values of sensors
+* Output : a single temperature value
+*/
+
+#include<stdio.h>
+
+#include "..\include\sdm.h"
+#include "..\include\eigen.h"
+#include "..\include\principle_comp.h"
+#include "..\include\contribution.h"
+
+int main()
+{
+int n = 4;
+double temperature[] = {2.75,3.55,3.70,4.0};
+
+double** sd_matrix;
+sd_matrix = (double**) malloc(n*sizeof(double));
+for(int r = 0; r < n; r++){
+    sd_matrix[r] = (double*) malloc(n*sizeof(double));
+}
+sd_matrix = support_degree_matrix(n,temperature);
+
+double** T;
+T = (double**) malloc(n*sizeof(double));
+for(int r = 0; r < n; r++){
+    T[r] = (double*) malloc(n*sizeof(double));
+}
+T = eigen_vector(n, sd_matrix);
+
+double* D;
```

Comment:
Bad variable name.

Reply:
I have created these variable names as they are in the given algorithm. So its easier to identify without any loss of generality.

Reply:
I disagree. If I read the code without the paper next to me, I have no idea what T is.

Reply:
Maybe. But for this specific project, I feel it's better to use variable names as specified in the algorithm because everyone including the reviewers and instructor is well-aware of the algorithm being implemented. Thanks for your suggestions though.